pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go

```diff
@@ -209,6 +210,14 @@ func (c *clusterMemberRemovalController) attemptToScaleDown(ctx context.Context,
 		}
 	}
 
+	// Don't scale down if all members are healthy and the cluster is already at the desired control-plane size
+	// When at the desired control-plane size, scale down is only allowed if a member is unhealthy so that a new machine can replace it.
+	// For healthy members, deleting a machine should result in a new machine being created and member addition before scale down can occur
+	if reflect.DeepEqual(liveVotingMembers, healthyLiveVotingMembers) && len(healthyLiveVotingMembers) <= desiredControlPlaneReplicasCount {
+		klog.V(2).Infof("skip scale down: all voting members are healthy and membership does not exceed the desired cluster size of %v", desiredControlPlaneReplicasCount)
```

Comment:
>alright, shall i use v(4)

Actually on second though this won't be the default case since a machine is pending deletion, but we're awaiting a new one to surge up the membership first. So we could keep it v(2).

But can you confirm what the operator's log show while 3 members are healthy with a machine pending deletion. Just want to gauge if this isn't too spammy.